### ISSUE TITLE
NEW FUNCTION: Send_Alerts.py - Send_All_Messages()

### ISSUE THREAD

**Issue by RwHendrickson** [RwHendrickson/AQ_SpikeAlerts#12]
See [workflow 2 - While Loop](https://docs.google.com/drawings/d/1-M7wfXUyXovk0JGWVRiF3X7jd6tNvfPP0M5g3lhY1PM/edit?usp=sharing)

This function(s?) should take in messages, record_ids_to_text, pg_connection_dict, and redCap_token_signUp & both Twilio tokens

then it will query REDCap for the record_ids' (from record_ids_to_text) phone numbers, message them their respective messages, and update our internal database with +1 to messages_sent and last_messaged = CURRENT_TIMESTAMP <- make sure it's in Central time

**Comment by RwHendrickson** [comment]
blackhat-hemsworth initialized 5db639f

Incorporated REDCap into script cf6b9299c940189660daea8a31db911863c2c88a

**Comment by RwHendrickson** [comment]
Notebook created to test Twilio functions. Minor tweaks (not yet added to scripts) but everything works

f14a8a5

Need to verify toll-free number to send URL. @blackhat-hemsworth  Any ideas on how to handle calls/TwiML functions?

**Comment by RwHendrickson** [comment]
blackhat-hemsworth finished up version 1 #17 

RwHendrickson cleaned up and incorporated into main 8aedffd

Will let it run for a day to see if we encounter any errors

**Comment by RwHendrickson** [comment]
The toll-free number is a bit of a headache. Waiting on verification, but they say that these numbers are not really for developers. The number works now so long as a URL is not sent, so worst case we can do this without linking to PurpleAir/the report. 

Twilio doesn't really offer any other options for developers at the moment, so we will just have to wait and see. 

Putting links here for future reference:

https://www.twilio.com/blog/developer-guide-to-toll-free-verification

**Comment by RwHendrickson** [comment]
Toll-Free Verification Approved! 

Whoa, wish I would've documented what I wrote. From memory, I remember I just put in my personal information and said we would be making brief Public Safety Announcements with a URL...

@blackhat-hemsworth do you think we're good to close this issue? Most of the other Twilio stuff I can think of probably condone creating a new issue, but would love to hear your thoughts!

**Comment by blackhat-hemsworth** [comment]
let's go!! 

I would say close this one, open a new one if any bugs come up & open a new one/two/three/etc. for those weirder functions like if we need to delete / scrub messages after people unsubscribe, or end up doing something with scheduled messages

**Comment by RwHendrickson** [comment]
Whoo!! Alright sounds good! Thanks so much again for your help 

The [Twilio scrub function](https://github.com/RwHendrickson/AQ_SpikeAlerts/blob/main/Scripts/python/twilio_functions.py) is actually already written (I snuck it in sometime last week) but the new issues I can think of (besides scheduling messages) are: webhook for responding to calls, personalized unsubscribed message, and applying for the [impact access program](https://support.twilio.com/hc/en-us/articles/4443965939995-Impact-Access-Program-Eligibility-Guidelines) as an Unregistered social impact initiative.

Non essential, tho. Yes, I think focus on some of the other stuff until an issue arises! 

Go team!!
